// @AxonPlataforma/design-system/index.js
/**
 * @AxonPlataforma/design-system v1.0.0
 *
 * Fuente unica de verdad para la identidad visual de Axon.
 * Todos los tokens, patrones y constantes del proyecto estan definidos aqui.
 *
 * USO:
 *   import { colors, typography, components } from '@AxonPlataforma/design-system';
 */

// ─────────────────────────────────────────────
// 1. BRAND
// ─────────────────────────────────────────────

export const brand = {
  name: 'AXON',
  fullName: 'AxonPlataforma',
  version: '1.0.0',
  tagline: 'Plataforma de Estudos Medicos',
  logo: {
    sizes: {
      xs:        { svg: 'w-3 h-[14px]', text: 'text-[9px] tracking-[0.25em]', gap: 'gap-1' },
      sm:        { svg: 'w-[14px] h-4',  text: 'text-[10px] tracking-[0.3em]', gap: 'gap-1.5' },
      md:        { svg: 'w-[18px] h-[22px]', text: 'text-[clamp(1rem,2vw,1.3rem)] tracking-[0.2em]', gap: 'gap-2' },
      lg:        { svg: 'w-6 h-7',       text: 'text-[clamp(1.2rem,2.5vw,1.8rem)] tracking-[0.25em]', gap: 'gap-2.5' },
      watermark: { svg: 'w-14 h-16',     text: 'text-[72px] tracking-[0.15em]', gap: 'gap-3' },
    },
    themes: {
      light:    { svgColor: 'text-white',    textColor: 'text-white/90' },
      dark:     { svgColor: 'text-gray-800', textColor: 'text-gray-800' },
      gradient: { svgColor: 'text-teal-500', textColor: 'text-teal-600' },
    },
  },
};

// ─────────────────────────────────────────────
// 2. COLORS
// ─────────────────────────────────────────────

export const colors = {
  /** Cor primaria de interacao — usada em botoes, links, icones ativos */
  primary: {
    50:  '#f0fdfa',   // bg-teal-50  — fundo de icones
    100: '#ccfbf1',   // bg-teal-100
    200: '#99f6e4',
    300: '#5eead4',
    400: '#2dd4bf',
    500: '#14b8a6',   // bg-teal-500 — botoes, badges, progresso
    600: '#0d9488',   // bg-teal-600 — botoes hover, texto ativo
    700: '#0f766e',   // bg-teal-700 — botoes pressed
    800: '#115e59',
    900: '#134e4a',
  },

  /** Sidebar e header escuro */
  dark: {
    navBar:       '#1e293b',   // Header top bar
    sidebarBase:  '#1c1c1e',   // Sidebar outer shell
    sidebarBody:  '#2d3e50',   // Sidebar content area
    cardDark:     '#2c3e50',   // Performance card, filter active
  },

  /** Backgrounds de pagina */
  surface: {
    page:       '#f9fafb',     // bg-gray-50  — fundo geral
    dashboard:  '#f5f2ea',     // fundo bege do dashboard
    card:       '#ffffff',     // bg-white — cards
    hover:      '#f3f4f6',     // bg-gray-100 — hover states
  },

  /** Bordas e divisores */
  border: {
    card:     '#e5e7eb',       // border-gray-200
    subtle:   'rgba(229,231,235,0.6)', // border-gray-200/60
    dark:     'rgba(255,255,255,0.1)', // border overlay on dark
    darkAlt:  'rgba(255,255,255,0.05)',
  },

  /** Texto */
  text: {
    primary:   '#111827',      // text-gray-900
    secondary: '#6b7280',      // text-gray-500
    tertiary:  '#9ca3af',      // text-gray-400
    disabled:  '#d1d5db',      // text-gray-300
    inverse:   '#ffffff',      // text-white
  },

  /** Semantic / status */
  semantic: {
    success:  '#10b981',       // emerald-500
    warning:  '#f59e0b',       // amber-500
    error:    '#ef4444',       // red-500
    info:     '#06b6d4',       // cyan-500
  },

  /** Mastery / spaced repetition levels */
  mastery: {
    notStarted: '#d1d5db',     // gray-300
    learning:   '#fbbf24',     // yellow-400
    reviewing:  '#14b8a6',     // teal-500
    mastered:   '#0d9488',     // teal-600
  },

  /** Rating scale (flashcard self-assessment) */
  ratings: {
    1: '#f43f5e',  // rose-500     — Nao sei
    2: '#f97316',  // orange-500   — Dificil
    3: '#facc15',  // yellow-400   — Medio
    4: '#84cc16',  // lime-500     — Facil
    5: '#10b981',  // emerald-500  — Perfeito
  },

  /** Cores de acento por disciplina (bg-*) */
  courseAccents: {
    anatomy:   { bg: 'bg-rose-400',   text: 'text-rose-400' },
    histology: { bg: 'bg-indigo-500', text: 'text-indigo-500' },
  },

  /** Chart palette */
  chart: {
    flashcards: '#14b8a6',     // teal-500
    videos:     '#06b6d4',     // cyan-500
    bar:        '#0d9488',     // teal-600
    grid:       '#f3f4f6',     // gray-100
  },
};

// ─────────────────────────────────────────────
// 3. TYPOGRAPHY
// ─────────────────────────────────────────────

export const typography = {
  /** Familias tipograficas */
  families: {
    heading: 'Georgia, serif',
    body:    '"Inter", system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, sans-serif',
    display: '"Space Grotesk", sans-serif',
    mono:    '"JetBrains Mono", monospace',
  },

  /** Google Fonts import URLs */
  imports: [
    'https://fonts.googleapis.com/css2?family=Inter:wght@300;400;500;600;700&display=swap',
    'https://fonts.googleapis.com/css2?family=Space+Grotesk:wght@300;400;500;600;700&family=JetBrains+Mono:wght@400;500&display=swap',
  ],

  /** Tailwind CSS variable name */
  cssVariable: '--font-sans',

  /** Regras de uso */
  rules: {
    pageTitle:     { family: 'heading', style: "fontFamily: 'Georgia, serif'", size: 'text-[clamp(2rem,4vw,3rem)]', weight: 'font-bold', tracking: 'tracking-tight' },
    sectionTitle:  { family: 'heading', style: "fontFamily: 'Georgia, serif'", size: 'text-lg',  weight: 'font-semibold' },
    sectionLabel:  { family: 'heading', style: "fontFamily: 'Georgia, serif'", size: 'text-sm',  weight: 'font-semibold', extra: 'uppercase tracking-wide' },
    cardTitle:     { family: 'heading', style: "fontFamily: 'Georgia, serif'", size: 'default',  weight: 'font-bold' },
    body:          { family: 'body',    style: "fontFamily: 'Inter'",          size: 'text-sm',  weight: 'font-medium' },
    caption:       { family: 'body',    style: "fontFamily: 'Inter'",          size: 'text-xs',  weight: 'font-medium' },
    label:         { family: 'body',    style: "fontFamily: 'Inter'",          size: 'text-[10px]', weight: 'font-medium', extra: 'uppercase tracking-wider' },
  },
};

// ─────────────────────────────────────────────
// 4. SHAPES & RADIUS
// ─────────────────────────────────────────────

export const shapes = {
  card:          'rounded-2xl',     // 16px — todos los cards
  button: {
    pill:        'rounded-full',    // botones de navegacion principales
    action:      'rounded-xl',      // botones de accion dentro de cards
    filter:      'rounded-lg',      // botones de filtro temporal
    filterAlt:   'rounded-md',      // filtros menores / tabs
  },
  icon:          'rounded-xl',      // contenedores de icono
  avatar:        'rounded-full',
  progress:      'rounded-full',    // barras de progresso
  input:         'rounded-xl',
  badge:         'rounded-full',
  sidebar: {
    navItem:     'rounded-xl',
    dropdown:    'rounded-2xl',
  },
};

// ─────────────────────────────────────────────
// 5. SHADOWS
// ─────────────────────────────────────────────

export const shadows = {
  card:        'shadow-sm',
  cardHover:   'shadow-md',
  chart:       'shadow-[0_2px_8px_rgba(0,0,0,0.04)]',
  performance: 'shadow-lg',
  dropdown:    'shadow-2xl',
  tooltip:     'shadow-[0_4px_12px_rgba(0,0,0,0.1)]',
};

// ─────────────────────────────────────────────
// 6. COMPONENT PATTERNS (Tailwind class recipes)
// ─────────────────────────────────────────────

export const components = {

  /** Patron de icono: fondo teal claro + icono teal */
  icon: {
    default:  { bg: 'bg-teal-50',    text: 'text-teal-500' },
    sizes: {
      sm: 'w-8 h-8',
      md: 'w-10 h-10',
      lg: 'w-12 h-12',
    },
    container: 'rounded-xl flex items-center justify-center',
    /** PROHIBIDO: no usar gradientes, no usar bg-blue-*, bg-violet-* */
  },

  /** Card generico */
  card: {
    base:     'bg-white rounded-2xl border border-gray-200 shadow-sm',
    hover:    'hover:shadow-md transition-shadow',
    padding:  'p-5',
    paddingLg: 'p-6',
    paddingXl: 'p-7',
  },

  /** Card oscuro (performance) */
  cardDark: {
    base:     'bg-[#2c3e50] rounded-2xl text-white shadow-lg',
    padding:  'p-7',
  },

  /** Boton primario (teal solido, pill) */
  buttonPrimary: {
    base:     'bg-teal-500 hover:bg-teal-600 text-white font-semibold rounded-full transition-colors',
    sizes: {
      sm:     'px-4 py-1.5 text-xs',
      md:     'px-6 py-2.5 text-sm',
      lg:     'px-8 py-3 text-base',
    },
  },

  /** Boton de accion dentro de cards */
  buttonAction: {
    base:     'bg-teal-600 hover:bg-teal-700 text-white font-semibold rounded-xl text-sm text-center transition-colors cursor-pointer',
    full:     'w-full py-2.5',
  },

  /** Filtro temporal */
  filterButton: {
    active:   'bg-teal-500 text-white shadow-sm',
    inactive: 'text-gray-500 hover:text-gray-700',
    /** Variante dark (WelcomeView) */
    activeDark:   'bg-[#2c3e50] text-white',
    inactiveDark: 'bg-gray-100 text-gray-600 hover:bg-gray-200',
  },

  /** Barra de progresso */
  progressBar: {
    track:    'w-full h-2 bg-gray-100 rounded-full overflow-hidden',
    fill:     'h-full rounded-full transition-all duration-700',
    colorDefault: 'bg-teal-500',
  },

  /** Progresso circular (SVG) */
  progressCircle: {
    strokeBg:     'rgba(255,255,255,0.1)',
    strokeActive: '#14b8a6',
    strokeWidth:  14,
    strokeWidthSmall: 3,
    strokeLinecap: 'round',
    strokeColorSmall: '#0d9488',
  },

  /** KPI Card */
  kpiCard: {
    base:     'bg-white p-5 rounded-2xl shadow-[0_2px_8px_rgba(0,0,0,0.04)] border border-gray-100 hover:shadow-md transition-all',
    iconBg:   'p-2.5 rounded-xl',
    trend: {
      up:     'text-green-700 bg-green-50',
      down:   'text-red-700 bg-red-50',
    },
  },

  /** AxonPageHeader */
  pageHeader: {
    wrapper:  'relative px-8 pt-4 pb-6 bg-white overflow-hidden border-b border-gray-200',
    title:    "text-[clamp(2rem,4vw,3rem)] font-bold text-gray-900 tracking-tight leading-[1]",
    titleFont: "Georgia, serif",
    subtitle: "text-sm text-gray-500",
    subtitleFont: "Georgia, serif",
  },

  /** Sidebar */
  sidebar: {
    width:        260,
    bgOuter:      '#1c1c1e',
    bgInner:      '#2d3e50',
    borderColor:  'rgba(255,255,255,0.05)',
    navItem: {
      base:     'w-full flex items-center gap-3 px-3 py-2.5 rounded-xl text-sm font-medium transition-all duration-200 group',
      active:   'bg-sky-500/10 text-sky-400 shadow-sm border border-sky-500/20',
      inactive: 'text-gray-400 hover:text-white hover:bg-white/5',
    },
    sectionLabel: 'px-3 text-xs font-bold text-gray-500 uppercase tracking-widest mb-2',
  },

  /** Header / Top Nav */
  header: {
    height:    'h-12',
    bg:        'bg-[#1e293b]',
    border:    'border-b border-white/10',
    menuBtn:   'p-2 bg-white/[0.08] hover:bg-white/15 border border-white/[0.1] rounded-lg text-white',
  },

  /** Course Switcher */
  courseSwitcher: {
    iconSize:     'w-10 h-10',
    iconRadius:   'rounded-lg',
    courseLabel:   'font-medium tracking-wide uppercase text-[14px] font-sans',
    courseName:   'font-semibold leading-none text-[16px] font-sans',
    dropdown:     'bg-white/80 backdrop-blur-xl border border-white/20 shadow-2xl rounded-2xl',
  },

  /** Quick access shortcut cards (WelcomeView) */
  shortcutCard: {
    base:     'bg-white rounded-2xl border border-gray-200 p-5 shadow-sm hover:shadow-md transition-shadow cursor-pointer',
    iconSize: 'w-10 h-10 rounded-xl',
    iconBg:   'bg-teal-50',
    iconColor: 'text-teal-500',
  },

  /** Course card (WelcomeView) */
  courseCard: {
    base:     'bg-white rounded-2xl border border-gray-200 p-5 shadow-sm hover:shadow-md transition-shadow',
    iconSize: 'w-12 h-12 rounded-xl',
    ctaButton: 'w-full py-2.5 rounded-xl bg-teal-600 hover:bg-teal-700 text-white text-sm font-semibold text-center transition-colors cursor-pointer',
  },

  /** Activity item */
  activityItem: {
    iconSize: 'w-10 h-10 rounded-lg',
    layout:   'flex items-start gap-3',
  },

  /** Chart containers */
  chartCard: {
    base:    'bg-white rounded-2xl p-6 shadow-[0_2px_8px_rgba(0,0,0,0.04)] border border-gray-100',
    tooltip: { borderRadius: '12px', border: 'none', boxShadow: '0 4px 12px rgba(0,0,0,0.1)' },
  },
};

// ─────────────────────────────────────────────
// 7. SECTION ACCENT COLORS (for content views)
// ─────────────────────────────────────────────

export const sectionColors = {
  /** Teal-only palette (ThreeDView, standardized) */
  teal: [
    { bg: 'from-teal-500 to-teal-600', light: 'bg-teal-50', text: 'text-teal-600', border: 'border-teal-200', dot: 'bg-teal-500', badge: 'bg-teal-100 text-teal-700' },
  ],

  /** Multi-color palette (FlashcardView, per-section) */
  multi: [
    { bg: 'from-teal-500 to-teal-600',    glow: 'shadow-teal-500/15',    light: 'bg-teal-50',    text: 'text-teal-600',    border: 'border-teal-200',    ring: 'ring-teal-500/20',    dot: 'bg-teal-500' },
    { bg: 'from-emerald-500 to-teal-600',  glow: 'shadow-emerald-500/15', light: 'bg-emerald-50', text: 'text-emerald-600', border: 'border-emerald-200', ring: 'ring-emerald-500/20', dot: 'bg-emerald-500' },
    { bg: 'from-sky-500 to-cyan-600',      glow: 'shadow-sky-500/15',     light: 'bg-sky-50',     text: 'text-sky-600',     border: 'border-sky-200',     ring: 'ring-sky-500/20',     dot: 'bg-sky-500' },
    { bg: 'from-amber-500 to-orange-600',  glow: 'shadow-amber-500/15',   light: 'bg-amber-50',   text: 'text-amber-600',   border: 'border-amber-200',   ring: 'ring-amber-500/20',   dot: 'bg-amber-500' },
  ],

  /** Disciplina-specific colors (WelcomeView course data) */
  disciplines: {
    microbiology: { icon: '\u{1F9A0}', iconBg: 'bg-purple-100', progressColor: 'bg-purple-500', percentColor: 'text-purple-600' },
    cellBiology:  { icon: '\u{1F33F}', iconBg: 'bg-teal-100',   progressColor: 'bg-teal-500',   percentColor: 'text-teal-600' },
    histology:    { icon: '\u{1F52C}', iconBg: 'bg-teal-100',   progressColor: 'bg-teal-500',   percentColor: 'text-teal-600' },
    anatomy:      { icon: '\u{2764}\u{FE0F}', iconBg: 'bg-pink-100',   progressColor: 'bg-pink-500',   percentColor: 'text-pink-600' },
  },
};

// ─────────────────────────────────────────────
// 8. NAVIGATION
// ─────────────────────────────────────────────

export const navigation = {
  /** All available views (must match ViewType in AppContext) */
  views: [
    'home',
    'dashboard',
    'study-hub',
    'study',
    'flashcards',
    'quiz',
    '3d',
    'schedule',
    'organize-study',
    'review-session',
    'study-dashboards',
    'knowledge-heatmap',
    'mastery-dashboard',
    'student-data',
  ],

  /** Primary sidebar items (icon names from lucide-react) */
  primaryItems: [
    { id: 'home',         label: 'Inicio',        icon: 'Home' },
    { id: 'dashboard',    label: 'Dashboard',      icon: 'LayoutDashboard' },
    { id: 'study-hub',    label: 'Estudar',        icon: 'BookOpen' },
    { id: 'schedule',     label: 'Cronograma',     icon: 'Calendar' },
    { id: 'flashcards',   label: 'Flashcards',     icon: 'Layers' },
    { id: '3d',           label: 'Atlas 3D',       icon: 'Box' },
    { id: 'quiz',         label: 'Quiz',           icon: 'GraduationCap' },
    { id: 'student-data', label: 'Meus Dados',     icon: 'Database' },
  ],

  /** Secondary sidebar items */
  secondaryItems: [
    { id: 'community', label: 'Comunidade',     icon: 'Users' },
    { id: 'settings',  label: 'Configuracoes',  icon: 'Settings' },
  ],

  /** Quick-access shortcuts (WelcomeView) */
  shortcuts: [
    { title: 'Resumos',    subtitle: 'Acessar resumos',     icon: 'BookOpen',       view: 'summaries' },
    { title: 'Flashcards', subtitle: 'Revisar cartoes',     icon: 'Layers',         view: 'flashcards' },
    { title: 'Quiz',       subtitle: 'Testar conhecimento', icon: 'GraduationCap',  view: 'quiz' },
    { title: 'Atlas 3D',   subtitle: 'Explorar modelos',    icon: 'Box',            view: '3d-atlas' },
  ],
};

// ─────────────────────────────────────────────
// 9. SPACING & LAYOUT
// ─────────────────────────────────────────────

export const layout = {
  sidebar: {
    width: 260,
    collapsedWidth: 0,
  },
  header: {
    height: 48, // h-12 = 3rem = 48px
  },
  content: {
    paddingX:  'px-6',
    paddingY:  'py-6',
    gap:       'gap-6',
    maxWidth:  'max-w-7xl',
  },
  grid: {
    stats:   'grid grid-cols-2 lg:grid-cols-4 gap-4',
    courses: 'grid grid-cols-1 lg:grid-cols-2 gap-5',
    kpi:     'grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-4 gap-4',
  },
  rightPanel: {
    width: 'w-[360px]',
  },
};

// ─────────────────────────────────────────────
// 10. ANIMATION
// ─────────────────────────────────────────────

export const animation = {
  pageTransition: {
    initial:  { opacity: 0, y: 10 },
    animate:  { opacity: 1, y: 0 },
    duration: 0.2,
  },
  cardEntrance: {
    initial:  { opacity: 0, y: 10 },
    animate:  { opacity: 1, y: 0 },
    staggerDelay: 0.1,
  },
  progressBar: {
    initial:  { width: 0 },
    duration: 1,
  },
  sidebar: {
    initial: { width: 260 },
    collapsed: { width: 0 },
  },
  dropdown: {
    initial:  { opacity: 0, y: 8, scale: 0.96 },
    animate:  { opacity: 1, y: 0, scale: 1 },
    exit:     { opacity: 0, y: 8, scale: 0.96 },
    duration: 0.15,
    ease:     'easeOut',
  },
};

// ─────────────────────────────────────────────
// 11. DESIGN RULES (reference for developers)
// ─────────────────────────────────────────────

export const designRules = {
  /** OBRIGATORIO */
  mandatory: [
    'Titulos SEMPRE em Georgia, serif (fontFamily inline style)',
    'Corpo SEMPRE em Inter (font-sans via Tailwind)',
    'Cor primaria de interacao: teal (#14b8a6)',
    'Botoes solidos: pill-shaped com rounded-full',
    'Cards brancos: rounded-2xl com shadow-sm',
    'Icones: bg-teal-50 + text-teal-500 (sem gradientes)',
    'Filtros ativos: bg-teal-500 text-white',
  ],

  /** PROIBIDO */
  forbidden: [
    'Glassmorphism (backdrop-blur em cards de conteudo)',
    'Gradientes em botoes ou icones',
    'Cores azul/violeta em elementos interativos (substituir por teal)',
    'bg-blue-*, bg-violet-*, bg-purple-* em botoes ou badges de acao',
    'Font-size via Tailwind classes (text-2xl, etc.) — usar clamp() ou tokens',
  ],

  /** Arquivo de componentes */
  fileStructure: {
    shared:  '/src/app/components/shared/',
    layout:  '/src/app/components/layout/',
    content: '/src/app/components/content/',
    ai:      '/src/app/components/ai/',
    ui:      '/src/app/components/ui/',
    context: '/src/app/context/',
    data:    '/src/app/data/',
    styles:  '/src/styles/',
  },
};

// ─────────────────────────────────────────────
// 12. BACKWARD COMPATIBILITY
// ─────────────────────────────────────────────

/** @deprecated Use `brand` instead */
export const BrandInfo = {
  name: brand.fullName,
  version: brand.version,
};
